presto-orc/src/main/java/com/facebook/presto/orc/cache/CachingOrcFileTailSource.java

```diff
@@ -41,11 +42,11 @@ public CachingOrcFileTailSource(OrcFileTailSource delegate, Cache<OrcDataSourceI
     }
 
     @Override
-    public OrcFileTail getOrcFileTail(OrcDataSource orcDataSource, MetadataReader metadataReader, Optional<OrcWriteValidation> writeValidation)
+    public OrcFileTail getOrcFileTail(OrcDataSource orcDataSource, MetadataReader metadataReader, Optional<OrcWriteValidation> writeValidation, HiveFileContext hiveFileContext)
             throws IOException
     {
         try {
-            return cache.get(orcDataSource.getId(), () -> delegate.getOrcFileTail(orcDataSource, metadataReader, writeValidation));
+            return cache.get(orcDataSource.getId(), () -> delegate.getOrcFileTail(orcDataSource, metadataReader, writeValidation, hiveFileContext));
```

Comment:
You need to test against `hiveFileContext` to see if you need to get data from cache or not. Check `CachingFileOpener` as an example